tests/test_main.py

```diff
@@ -1,186 +1,165 @@
 import uuid
+from unittest import mock
+
 import pytest
-from fastapi import HTTPException
-from fastapi.testclient import TestClient
-from src.main import app
 from typing import Generator
-from src.main import (
-    projects,
-    get_project,
-    create_project_,
-    get_projects_,
-    update_project_details_,
-    delete_project_,
-)
-from src.schemas import Project, ProjectDetails
+from unittest.mock import patch, MagicMock
+from fastapi.testclient import TestClient
+import os
 
 
-@pytest.fixture
-def client() -> Generator[TestClient]:
-    yield TestClient(app)
+os.environ.update(
+    {
+        "POSTGRES_USER": "test_user",
+        "POSTGRES_PASSWORD": "test_password",
+        "POSTGRES_DB": "test_db",
+        "POSTGRES_PORT": "5432",
+        "POSTGRES_SERVER": "localhost",
+        "DATABASE_URL": "postgresql://test_user:test_password@localhost:5432/test_db",
+    }
+)
```

Comment:
this is not a good practice. You modify mutable value during tests, it affects other tests. What if you decide to add integration tests with real db?
Please use patch and fixture for that. Google how people patch env vars in python for tests